### ISSUE TITLE
make use of PETSc MATPRODUCT

### ISSUE THREAD

**Issue by haplav** [It4innovations/permon#44]
We should abandon our own implementation in favor of PETSc one, which seems to be under intensive development now (https://gitlab.com/petsc/petsc/-/merge_requests/2800).

If there is anything useful on our side, it should be contributed to PETSc.

**Comment by jkruzik** [comment]
Do you mean https://github.com/permon/permon/blob/master/src/mat/impls/composite/matprod.c ?
MATPRODUCT in PETSc is basically an interface for various MatMatMults, while our MatProd is a convenience over MatComposite.

**Comment by haplav** [comment]
Yes, you are right. But we also use our MatProd for a product of 3 matrices. Implicit, though. Maybe it should cover also the implicit product if it's not the case yet.

Anyway, our improvements to MatProd/MatSum should be also contributed. Including splitting into two classes. MatComposite doesn't make sense. Sum and product are two completely different things.

**Comment by jkruzik** [comment]
MatProd/MatSum classes were removed in v3.21.0 (ce39723, 0935983) in favour of MatComposite after MatComposite gained the functionality we were missing.

Splitting MatComposite into two classes can be suggested in PETSc. I don't have a strong preference.
